contrib/babelfishpg_tsql/sql/sys_views.sql

```diff
@@ -2981,3 +2981,56 @@ LEFT JOIN pg_foreign_server AS f ON u.srvid = f.oid
 LEFT JOIN pg_foreign_data_wrapper AS w ON f.srvfdw = w.oid
 WHERE w.fdwname = 'tds_fdw';
 GRANT SELECT ON sys.linked_logins TO PUBLIC;
+
+CREATE OR REPLACE FUNCTION sys.babelfish_get_extended_properties()
+RETURNS TABLE (
+  class sys.tinyint,
+  class_desc sys.nvarchar(60),
+  major_id int,
+  minor_id int,
+  name sys.sysname,
+  value sys.sql_variant
+)
+AS $$
+DECLARE
+  rec record;
+  class sys.tinyint;
+  class_desc sys.nvarchar(60);
+  major_id int;
+  minor_id int;
+BEGIN
+  FOR rec IN SELECT * FROM sys.babelfish_extended_properties WHERE dbid = sys.db_id()
```

Comment:
I suspect this implementation would be very slow when large amount of objects exists. 
Why not just use a subquery?

Reply:
sorry, I don't really know how to use a subquery. Do you mean 'case when'?

Reply:
Yes, just define the view with one single select statement with "case when"